### ISSUE TITLE
Support signed-pages add-on

### ISSUE THREAD

**Issue by rugk** [PrivateBin/PrivateBin#302]
Folow-up of https://github.com/PrivateBin/PrivateBin/issues/138#issuecomment-362920502

Requirements:
1. HTML code delivered must be static.
2. Adjustments in the config should be fetched via JSON calls or so after loading and elements may dynamically be hidden.
3. Change the API to use a subdir (`/api`) or so. As it currently [uses the main site](https://github.com/PrivateBin/PrivateBin/wiki/API) that may not be good.
   Because when this is done, we can serve a HTML-only file fir index and we can provide a real API. (Maybe we call it REST? :wink: Would need to look into how it would have to be designed to be real REST.)
4. Get added to https://github.com/tasn/webext-signed-pages.

**Comment by elrido** [comment]
We currently distinguish between browser and API calls based on the `X-Requested-With: JSONHttpRequest` header in HTTP, so it is very clear to the end-point if a request is for the API or not. Using a different URL would mainly help to distinguish between these in the access log file of the webserver.

**Comment by rugk** [comment]
Yeah, and thus likely also makes configuring the web server easier, if you e.g. want to disallow some stuff of so.

**Comment by rugk** [comment]
FYI I just saw the probably only use if the "signed pages" extension is EteSync in their new beta web app, apparently: https://pim.etesync.com/

Their [deployment script here](https://github.com/etesync/etesync-web/blob/master/deploy.sh) apparently [uses](https://github.com/etesync/etesync-web/commit/bc4e0224505e8f1da1316c5982e0396e9d0d9e46) [an official JS script for signing](https://github.com/tasn/webext-signed-pages?tab=readme-ov-file#as-a-developer).
